extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -24,27 +24,29 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 	private _throttleTimer: any;
 	private _resource: string;
 
-	constructor(private workspaceRoot: string, extensionContext: vscode.ExtensionContext) {
-		if (workspaceRoot !== '') {
-			this._tableOfContentsPath = this.getTocFiles(this.workspaceRoot);
-			let bookOpened: boolean = this._tableOfContentsPath && this._tableOfContentsPath.length > 0;
-			vscode.commands.executeCommand('setContext', 'bookOpened', bookOpened);
-		}
+	constructor(workspaceFolders: vscode.WorkspaceFolder[], extensionContext: vscode.ExtensionContext) {
+		let workspacePaths: string[] = workspaceFolders.map(a => a.uri.fsPath);
+		this._tableOfContentsPath = this.getTableOfContentFiles(workspacePaths);
```

Comment:
Can we move the logic here (getting table of content files and setting the bookOpened variable) to an async method? We could just kick it off here and catch any errors in the thenable function.

This can be done as a separate PR, but why I'm asking is 
1. The constructor here operates in a sync fashion even while scanning files on disk. This should be discouraged: ideally what we'd do is move all file reads to the `getChildren` or `getBooks` method. Given we need the context key I understand why we can't wait, but let's move as async so at least overall extension startup isn't blocked on this.
2. We need to cover the case where files / folders are updated in the workspace. This can be by:
  - adding/removing workspace folders (listen to `vscode.workspace.onDidChangeWorkspaceFolders`), in which case books will appear / be removed,
  -  OR by updating the TOC file (use `vscode.workspace.createFileSystemWatcher` and filter to _toc.yml files for efficient management). The scenario here is that book sections will change (be added / removed) from an existing book. 
In both cases, we should clear the cached books object and fire an`onDidChangeTreeData` event to re-assess the content.
I would recommend addressing these in a separate PR, but let's create an issue now and plan when to do this. I think we should aim to get something in this week if possible.